npm_and_yarn/lib/dependabot/npm_and_yarn/package_name.rb

```diff
@@ -38,6 +45,17 @@ def eql?(other)
         to_s.eql?(other.to_s)
       end
 
+      def library_name
+        return self unless types_package?
+
+        @library_name ||=
+          if scoped_types_package?
+            self.class.new(unscoped_types_package_name)
+          else
+            self.class.new(@name.to_s)
+          end
+      end
```

Comment:
If you adopt the `TYPES_PACKAGE_NAME_REGEX` change above, you can avoid helper methods altogether:

```suggestion
      def library_name
        @library_name ||= 
          if match = TYPES_PACKAGE_NAME_REGEX.match(to_s)
            if match[:scope]
              self.class.new("@#{match[:scope]}/#{match[:name]}")
            else
              self.class.new(match[:name].to_s)
            end
          end
      end
```

(Not sure if RuboCop will allow `if` with assignment. You may have to do `match =` and `if match` on separate lines)

Reply:
👍 All comments addressed together in [14bb063](https://github.com/dependabot/dependabot-core/pull/5025/commits/14bb0637d9a455e903ec498a4291c88c54011571)